src/livekit/MediaDevicesContext.tsx

```diff
@@ -59,24 +63,73 @@ export interface MediaDevice {
   select: (deviceId: string) => void;
 }
 
-export interface MediaDevices {
-  audioInput: MediaDevice;
-  audioOutput: MediaDevice;
-  videoInput: MediaDevice;
+interface InputDevices {
+  audioInput: MediaDeviceHandle;
+  videoInput: MediaDeviceHandle;
   startUsingDeviceNames: () => void;
   stopUsingDeviceNames: () => void;
+  usingNames: boolean;
 }
 
-function useMediaDevice(
+export interface MediaDevices extends Omit<InputDevices, "usingNames"> {
+  audioOutput: MediaDeviceHandle;
+}
+
+/**
+ * An observable that represents if we should display the devices menu for iOS.
+ * This implies the following
+ *  - hide any input devices (they do not work anyhow on ios)
+ *  - Show a button to show the native output picker instead.
+ *  - Only show the earpiece toggle option if the earpiece is available:
+ *   `setAvailableOutputDevices$.includes((d)=>d.forEarpiece)`
+ */
+export const iosDeviceMenu$ = alwaysShowIphoneEarpieceSetting.value$.pipe(
+  startWith(
+    alwaysShowIphoneEarpieceSetting.getValue() ||
+      navigator.userAgent.includes("iPhone"),
+  ),
```

Comment:
This `startWith` should be redundant since `Setting["value$"]` refers to a behavior-like observable (an observable that emits immediately upon subscription)

Reply:
I got tricked by it being stored in an Observable. Which I now notice is just a trick to hide the `.next` "write" api and making this a readonly behaviour subject basically.
But since it references to the BahaviourSubject its still inheriting the "emit on subscription" property.